### PR TITLE
fix(test): replace Bun.sleep(0) with Promise.resolve() in opencode-process.spec.ts (fixes #1634)

### LINKED ISSUE #1634: fix(test): Bun.sleep(0) comment accuracy in opencode-process.spec.ts
## Background

PR #1628 replaced `await new Promise((r) => setTimeout(r, 10))` with `await Bun.sleep(0)` in `packages/opencode/src/opencode-process.spec.ts` at lines 181 and 220, commenting this as "flush microtasks". Copilot comments #3133804269 and #3133804284 on PR #1628 raised this during review.

## Problem

`await Bun.sleep(0)` yields via a timer/event-loop turn, **not** a pure microtask flush. The comment "Flush microtasks so the .then() handler fires" is technically inaccurate and may mislead future readers into thinking this is a lightweight operation. If the test only needs a microtask flush, `await Promise.resolve()` (or multiple chained `await`s) would be the correct primitive.

## Proposed Fix

Determine which tests need a genuine microtask flush vs. an event-loop turn:
- If `.then()` callback doesn't need an I/O or timer event, use `await Promise.resolve()` instead of `Bun.sleep(0)`
- Update comments to accurately describe the mechanism

## Files
- `packages/opencode/src/opencode-process.spec.ts:181,220`

### PR DESCRIPTION
**Pull request by theshadow27**
## Summary
- `Bun.sleep(0)` yields an event-loop tick (timer/macrotask), not a microtask flush — wrong primitive for waiting on `.then()` reactions
- Both sites replaced with `await Promise.resolve()`, which is the correct primitive: it flushes the microtask queue and the `.then()` reaction on `exited` fires before the await resumes
- Comments updated from "Yield to the event loop..." to "Flush microtask queue so the .then() reaction on exited fires." to accurately describe the mechanism

## Test plan
- [ ] All 16 tests in `opencode-process.spec.ts` pass (`bun test packages/opencode/src/opencode-process.spec.ts`)
- [ ] Full pre-commit suite (typecheck + lint + coverage) passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/opencode/src/opencode-process.spec.ts [hunk 1]
```diff
@@ -177,8 +177,8 @@ describe("OpenCodeProcess", () => {
     });
 
     await proc.spawn();
-    // Yield to the event loop so the exited promise reaction chain settles before asserting.
-    await Bun.sleep(0);
+    // Flush microtask queue so the .then() reaction on exited fires.
+    await Promise.resolve();
     expect(exitCode).toBe(0);
```
[the remark this answers is not in the document]
Comment (theshadow27) [reply packages/opencode/src/opencode-process.spec.ts:183]: Fixed in c33ec6c — destructured `result` from `mockSpawn` and replaced `await Promise.resolve()` with `await result.exited`. Same promise the process monitors, so our continuation is always enqueued after the `.then()` reaction already registered in `spawn()` — deterministic regardless of microtask queue depth.

File: packages/opencode/src/opencode-process.spec.ts [hunk 2]
```diff
@@ -216,8 +216,8 @@ describe("OpenCodeProcess", () => {
 
     // Trigger exit
     resolveExited(0);
-    // Yield to the event loop so the exited promise reaction chain settles before asserting.
-    await Bun.sleep(0);
+    // Flush microtask queue so the .then() reaction on exited fires.
+    await Promise.resolve();
```
[the remark this answers is not in the document]
Comment (theshadow27) [reply packages/opencode/src/opencode-process.spec.ts]: Fixed in c33ec6c — replaced `await Promise.resolve()` with `await exitedPromise` (already in scope). Process registers its `.then()` before this await attaches, so it always runs first — fully deterministic.

**Comment by theshadow27** [comment]
## QA Verification

**PR:** #2001 — fix(test): replace Bun.sleep(0) with Promise.resolve() in opencode-process.spec.ts (fixes #1634)
**Linked issue:** #1634

### Implementation Evidence
- `packages/opencode/src/opencode-process.spec.ts:182` — `await Bun.sleep(0)` replaced with `await result.exited` (awaiting the same promise the process monitors; more deterministic than `Promise.resolve()`)
- `packages/opencode/src/opencode-process.spec.ts:222` — `await Bun.sleep(0)` replaced with `await exitedPromise` for same reason
- Comments updated to accurately describe the mechanism: "Await the same promise the process monitors so our continuation is enqueued after the .then() reaction in spawn() — deterministic regardless of microtask depth."
- Note: PR description mentions `Promise.resolve()` but the actual implementation uses the stronger `await <same-promise>` idiom, which is correct and more robust

### Test Coverage
- `packages/opencode/src/opencode-process.spec.ts` — 16 tests covering spawn, onExit callback, double-call guard, kill, URL discovery, and edge cases. Both modified tests pass cleanly.

### Test Results
16 pass, 0 fail (63ms)

### CI Status
All checks green:
- `build` — pass (7s)
- `check` — pass (4m6s)
- `coverage` — pass (6m6s)

### Copilot Threads
2 Copilot inline threads, both already replied to by `theshadow27`. No CHANGES_REQUESTED reviews.

### Remaining Work (non-blocking)
None identified.

### Outcome
**qa:pass** — Both `Bun.sleep(0)` sites correctly replaced with deterministic promise awaits. Tests and CI green. Copilot threads addressed.
